### DISCUSSION TITLE
Drop target Bank with switch matrix output to Cobrapin with direct inputs

### DISCUSSION THREAD

**Discussion by Stefku** [missionpinball/missionpinball#89] [Q&A] [answered]
Hi there

I bought a drop target bank that has pins for output for switch matrix, having pins. three rows, one col.
I want to connect it to cobrapin with direct input, thus no switch matrix.

After googling around I thought I have to connect cobrapin ground to col and cobrapin switch input to row. But nothing happens. I also tried vice versa row to ground and col to cobrapin switch input.

Does anybody have a hint?

Cheers
Stefan

PS: this is the drop target: https://www.pinballlife.com/mm5/merchant.mvc?Screen=PROD&Product_Code=PBL-100-0078-00

**Comment by ralfit0** [comment, marked as answer]
Stefan,
**IF** the below is the correct schematic, direct switches with Cobra GND connected to the Opto board's COL and Cobra's switch pins to Opto pins 1, 2, 3 should work fine. If the schematic is not the right one, be aware of this quote:
_WARNING: the switch inputs should be limited to 3.3V since the STM32 is a 3.3V device. If you are using an active switch like a drop target opto board, make sure none of the switch pins can go above 3.3V._ 

Have you powered the optos, though? 12VDC + on pin 7, - on pin 8 ?
![optobrd](https://user-images.githubusercontent.com/99437486/227028891-8acd69ee-d435-4b81-afda-eb1ec6bb642b.jpg)
Ralf

**Reply by Stefku** [reply]
Hi Ralf, thanks for you answer and the schematics. With that I set up a test with a voltage meter. When I put 5V to ROW 1 and measure the voltage between COL and GND I see the open and closed state of the optos.
Problem is, I cannot connect directly to cobrapin inputs, because I just have one signal output namely COL. But that's how matrix work, as far as I understand. The rows are powered separately and the matrix logic knows which switch is closed.

So if I don't miss something. Either I create a logic that converts matrix behaviour in "parallel" behavior, or I just connect another PCB that is "parallel" by design.

PS: You're absolutely right, I have to limit the output signal to 3.3V
PPS: I used 5V instead of 12V that worked fine. Maybe 3.3V ist also sufficient? I try.

**Comment by ralfit0** [comment]
Stefan,
"When I put 5V to ROW 1 and measure the voltage between COL and GND I see the open and closed state of the optos."
- that leaves me a bit clueless: You should have all DC grounds in your system connected. Not sure why you are able to measure a opening/closing Opto when you connect GND with COL (= switch GND). It should not be like that.

Back to what should work: a "1 switch" matrix is not so much different than direct switch wiring. (The signal on the line is a different story). Power the opto board (if 5VDC works: great) with pins 6 & 7. Connect Pin 4 (COL) of the Opto board to e.g. Cobra J3, pin 5 ("GND") and Pin 1 (ROW 1) of the Opto board to e.g. Cobra J3, pin 6 ("1-0-8"). Configure a 1-0-8 switch in MPF as NC (normally closed). It should trigger now, if you activate opto1 on the board.

All of the above only if the schematic is the correct one (which I cannot say, as I do not have that target bank myself)!
Ralf

**Reply by Stefku** [reply]
Thanks Ralf, it works as you mentioned in your first answer!

I always thought I have to power the opto receiver as well, but it works like a normal switch. At least these ones on the board. I designed all my custom PCBs for Optos kind of too complicated.. :-D

**Reply by ralfit0** [reply]
Stefan, glad it works for you now!

**Reply by cobra18t** [reply]
Did you end up connecting it in the matrix or as direct switches in the end?

**Reply by Stefku** [reply]
With direct switches

**Reply by cobra18t** [reply]
OK. For reference, that one should also work with matrix switches. Optos are slower to respond in a matrix, so it may require the stronger pull-ups on the affected rows: https://pinballmakers.com/wiki/index.php?title=CobraPin#Matrix_Row_Pull_Resistor_Jumpers
